### PR TITLE
Docs: CSRF add configuration options and documentation for additional headers and origins

### PR DESCRIPTION
**Pull request by eleijonmarck**
**What this PR does / why we need it**:
This adds documentation for handling csrf checks that will allow customers to add additional headers or include trusted origins to bypass csrf checks.

Original PR: https://github.com/grafana/grafana/pull/48120  

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by sakjur — CHANGES_REQUESTED** [review]
The options are in the security section, not auth.saml.

**Comment by eleijonmarck** [comment]
@sakjur too much SAML, moved it to `[security]`

File: docs/sources/setup-grafana/configure-security/configure-authentication/saml.md [hunk 1]
```diff
@@ -407,3 +407,20 @@ The keys you provide should look like:
...
+When the user logs in using SAML and gets presented with "origin not allowed", the user might be issuing the login from an IdP (identity provider) service or the user is behind a reverse proxy. This potentially happens as Grafana's CSRF checks deem the requests to be invalid. For more information [CSRF](https://en.wikipedia.org/wiki/Cross-site_request_forgery).
+
+To solve this issue, you can configure either the [`csrf_trusted_origins`]({{< relref "../../configure-grafana/enterprise-configuration/#csrf-trusted-origins" >}}) or [`csrf_origin_headers`]({{< relref "../../configure-grafana/enterprise-configuration/#csrf-origin-headers" >}}) option in the SAML configuration.
+
+Example of a configuration file:
+
+```bash
+# config.ini
+...
+[auth.saml]
```
Comment (sakjur) [inline docs/sources/setup-grafana/configure-security/configure-authentication/saml.md:422]:
```suggestion
[security]
```

File: docs/sources/setup-grafana/configure-security/configure-authentication/saml.md [hunk 2]
```diff
@@ -407,3 +407,20 @@ The keys you provide should look like:
 ...
 -----END PRIVATE KEY-----
 ```
+
+### SAML login attempts fail with request response "origin not allowed"
+
+When the user logs in using SAML and gets presented with "origin not allowed", the user might be issuing the login from an IdP (identity provider) service or the user is behind a reverse proxy. This potentially happens as Grafana's CSRF checks deem the requests to be invalid. For more information [CSRF](https://en.wikipedia.org/wiki/Cross-site_request_forgery).
```
Comment (sakjur) [inline docs/sources/setup-grafana/configure-security/configure-authentication/saml.md:413]:
```suggestion
When the user logs in using SAML and gets presented with "origin not allowed", the user might be issuing the login from an IdP (identity provider) service or the user is behind a reverse proxy. This potentially happens as Grafana's CSRF checks deem the requests to be invalid. For more information [CSRF](https://owasp.org/www-community/attacks/csrf).
```

File: conf/sample.ini [hunk 3]
```diff
@@ -307,6 +307,13 @@
 
 # Controls if old angular plugins are supported or not. This will be disabled by default in future release
 ;angular_support_enabled = true
+;
```
Comment (sakjur) [inline conf/sample.ini:310]:
```suggestion
```

**Review by sakjur — APPROVED** [review]
Looks mostly good from my POV now. See comments for specific suggestions (the first one I expect you to apply before merging, the other two are more matters of preference)

I’ll let a technical writer take care of any add’t review 😊

File: docs/sources/setup-grafana/configure-security/configure-authentication/saml.md [hunk 4]
```diff
@@ -407,3 +407,20 @@ The keys you provide should look like:
...
+To solve this issue, you can configure either the [`csrf_trusted_origins`]({{< relref "../../configure-grafana/enterprise-configuration/#csrf-trusted-origins" >}}) or [`csrf_origin_headers`]({{< relref "../../configure-grafana/enterprise-configuration/#csrf-origin-headers" >}}) option in the SAML configuration.
+
+Example of a configuration file:
+
+```bash
+# config.ini
+...
+[auth.saml]
+csrf_trusted_origins = https://grafana.example.com
+csrf_origin_headers = X-Forwarded-Proto X-Forwarded-Host X-Forwarded-Port
```
Comment (heoelri) [inline docs/sources/setup-grafana/configure-security/configure-authentication/saml.md:424]: Is `csrf_origin_headers` the correct option to use here? `csrf.go` uses only `csrf_additional_headers` and `csrf_trusted_origins`.

https://github.com/grafana/grafana/blob/d32ae1890971343fe24ea9bf098bca049992c3dd/pkg/middleware/csrf/csrf.go#L38-L39

Comment (eleijonmarck) [reply docs/sources/setup-grafana/configure-security/configure-authentication/saml.md:424]: it is not! thanks for pointing it out!

Comment (eleijonmarck) [reply docs/sources/setup-grafana/configure-security/configure-authentication/saml.md:424]: https://github.com/grafana/grafana/pull/50589

**Pull request merged**